packages/utils/src/timeout.ts

```diff
@@ -1,23 +1,35 @@
-import {anySignal} from "any-signal";
-import {TimeoutError} from "./errors.js";
-import {sleep} from "./sleep.js";
+import { TimeoutError } from "./errors.js";
+import { sleep } from "./sleep.js";
 
 export async function withTimeout<T>(
   asyncFn: (timeoutAndParentSignal?: AbortSignal) => Promise<T>,
   timeoutMs: number,
   signal?: AbortSignal
 ): Promise<T> {
   const timeoutAbortController = new AbortController();
-  const timeoutAndParentSignal = anySignal([timeoutAbortController.signal, ...(signal ? [signal] : [])]);
+  const timeoutAndParentSignal = timeoutAbortController.signal;
+
+  let abortListener: (() => void) | undefined;
+
+  if (signal) {
+    abortListener = () => timeoutAbortController.abort();
+    signal.addEventListener("abort", abortListener);
+  }
```

Comment:
If you pass `signal` property as third parameter of `addEventListner` then you don't need to create to cleanup the event handler later on. And that will also remove the need of having `abortListener` a separate variable. 

https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#signal